piff/kernel.py

```diff
@@ -0,0 +1,293 @@
+                # dL_ij/dth_k is a matrix with all zeros except for one element.  That element is
+                # L_ij if k indicates one of the theta parameters landing on the Cholesky diagonal,
+                # and is 1.0 if k indicates one of the thetas in the lower triangular region.
+                L_grad = np.zeros((self.ntheta, self.ndim, self.ndim), dtype=float)
+                L_grad[(np.arange(self.ndim),)+self._d] = self._L[self._d]
+                L_grad[(np.arange(self.ndim, self.ntheta),)+self._t] = 1.0
+
+                half_invLam_grad = np.dot(L_grad, self._L.T)
+                invLam_grad = half_invLam_grad + np.transpose(half_invLam_grad, (0, 2, 1))
+
+                dX = X[:, np.newaxis, :] - X[np.newaxis, :, :]
+                dist_grad = np.einsum("ijk,lkm,ijm->ijl", dX, invLam_grad, dX)
+                K_gradient = -0.5 * K[:, :, np.newaxis] * dist_grad
+                return K, K_gradient
+        else:
+            return K
+
+    @property
+    def hyperparameter_cholesky_factor(self):
+        return Hyperparameter("CholeskyFactor", "numeric", (1e-5, 1e5), int(self.ntheta))
+
+    def get_params(self, deep=True):
+        return {"invLam":self.invLam}
+
+    def set_params(self, invLam=None):
+        if invLam is not None:
+            self.invLam = invLam
+            self._L = np.linalg.cholesky(self.invLam)
+            self._theta = np.hstack([np.log(self._L[self._d]), self._L[self._t]])
+
+    @property
+    def theta(self):
+        return self._theta
+
+    @theta.setter
+    def theta(self, theta):
+        self._theta = theta
+        self._L = np.zeros_like(self.invLam)
+        self._L[np.diag_indices(self.ndim)] = np.exp(theta[:self.ndim])
+        self._L[np.tril_indices(self.ndim, -1)] = theta[self.ndim:]
+        self.invLam = np.dot(self._L, self._L.T)
+
+    def __repr__(self):
+        return "{0}(invLam={1!r})".format(self.__class__.__name__, self.invLam)
+
+    @property
+    def bounds(self):
+        return self._bounds
+
+
+class VonKarman(StationaryKernelMixin, NormalizedKernelMixin, Kernel):
+    """VonKarman kernel.
+
+    Parameters
+    -----------
+    length_scale : float or array with shape (n_features,), default: 1.0
+        The length scale of the kernel. If a float, an isotropic kernel is
+        used. If an array, an anisotropic kernel is used where each dimension
+        of l defines the length-scale of the respective feature dimension.
+
+    length_scale_bounds : pair of floats >= 0, default: (1e-5, 1e5)
+        The lower and upper bound on length_scale
+    """
+    def __init__(self, length_scale=1.0, length_scale_bounds=(1e-5, 1e5)):
+        self.length_scale = length_scale
+        self.length_scale_bounds = length_scale_bounds
+
+    @property
+    def anisotropic(self):
+        return np.iterable(self.length_scale) and len(self.length_scale) > 1
+
+    @property
+    def hyperparameter_length_scale(self):
+        if self.anisotropic:
```

Comment:
You don't seem to test this option.  Should add a test that hits these branches.

Reply:
I do not used it for the moment, because I want to use it latter, when I will add option of fitting anisotropic kernel with two-point correlation function. Shall I remove it or can I let it like that for the moment ?